edu-services/gradebook-service/hibernate/src/java/org/sakaiproject/tool/gradebook/Assignment.java

```diff
@@ -241,7 +241,16 @@ public int compare(Object o1, Object o2) {
                 // if categories are null
                 if (one.getCategory() == null && two.getCategory() == null) {
                     // sort by assignment sort order
-                    return one.getSortOrder().compareTo(two.getSortOrder());
+                    if (one.getSortOrder() == null && two.getSortOrder() == null) {
```

Comment:
I've added a check for `getSortOrder` nulls and revert to the Id to get some kind of sorting.